libcudacxx/include/cuda/__cmath/ceil_div.h

```diff
@@ -58,6 +59,11 @@ ceil_div(const _Tp __a, const _Up __b) noexcept
   {
     return static_cast<_Common>((__a1 + __b1 - 1) / __b1);
   }
+  else if (_CUDA_VSTD::is_constant_evaluated())
+  {
+    const auto __res = __a1 / __b1;
+    return static_cast<_Common>(__res + (__res * __b1 != __a1));
+  }
```

Comment:
I would prefer if we would nest that inside the `else` branch, I personally do not like mixing if constexpr with "ordinary" if